base/range.jl

```diff
@@ -469,9 +469,10 @@ A range `r` where `r[i]` produces values of type `T` (in the first
 form, `T` is deduced automatically), parameterized by a `ref`erence
 value, a `step`, and the `len`gth. By default `ref` is the starting
 value `r[1]`, but alternatively you can supply it as the value of
-`r[offset]` for some other index `1 <= offset <= len`. In conjunction
-with `TwicePrecision` this can be used to implement ranges that are
-free of roundoff error.
+`r[offset]` for some other index `1 <= offset <= len`. The syntax `a:b`
+or `a:b:c` with `a`, `b`, and `c` all floating-points creates a `StepRangeLen`.
```

Comment:
```suggestion
or `a:b:c`, where any of `a`, `b`, or `c` are floating-point numbers, creates a
`StepRangeLen`.
```
They don’t have to be _all_ floating-point numbers, one of them is enough. `1:2.0`, `1.0:2:3`, `1:2.0:3` all create `StepRangeLen`s.

I also changed "floating-points" to "floating-point numbers".

Reply:
Maybe the first question is whether this should be documented at all. Or is the exact type of range you get just an implementation detail?

Reply:
I would actually say it is an implementation detail that shouldn’t be documented. Especially because anyone can define their own datatypes, for which `a:b:c` can return whatever they want.